amazon-speech-api/amazonTranscribeHarris.js

```diff
@@ -2,29 +2,112 @@ const request = require('request')
 const AWS = require('aws-sdk');
 AWS.config.update({region:'us-east-1'});
 const transcribeservice = new AWS.TranscribeService();
+const s3 = new AWS.S3();
 
-const params = {
+function generateRandomString() {
+  return Math.random().toString(36).substring(7);
+}
+let fetchedVocabulary = false;
+
+let transcriptionObject = {
   file: 'https://s3.amazonaws.com/talk-long-example/REC-40.-Driving.mp3',
+  createBucket: generateRandomString(),
   format: 'mp3',
-  jobName: 'transcribe-anson-voice-recording-test-11',
+  jobName: generateRandomString(),
   outputBucketName: 'talk-long-example',
-  outputJson:'https://s3.amazonaws.com/talk-long-example/transcribe-anson-voice-recording-test-10.json',
 };
 
-function createVocabulary() {
-  const params = {
-    LanguageCode: 'en-US',
-    Phrases: ['jobs','are','good'],
-    VocabularyName: 'harris_test_amazon_transcribe' /* required */
-  };
-  // NOTE: You can only call this function once.  Afterwards, the VocabularyName is already taken
-  transcribeservice.createVocabulary(params, function(err, data) {
-    if (err) console.log(err, err.stack); // an error occurred
-    else     console.log(data);           // successful response
-  });
+let vocabularyObject = {
+  LanguageCode: 'en-US',
+  Phrases: ["candidates", "talkhiring","jobseekers","resume"], // Array of Strings
```

Comment:
do you know how the vocabulary works in aws transcribe?  my main question is: are these the only words that aws can transcribe now?

Reply:
Well the aws createVocabulary phrases request requires a list containing transcribe words that are typically non-English or specific words it may have otherwise not recognized. Transcribe still functions normally except now adjusting to words in the phrases. As for the phrases, I remembered when we tested tested google's transcribe, these were some words that gave frequent errors. I have not as of yet came across a problem with phrasing yet so I choose an arbitrary list.

Reply:
I tested to see whether the transcription would use the giving vocabulary. I forgot to test how well it performs on the transcription with the vocabulary , but I will be testing it either today or tomorrow